src/System.Memory/ref/System.Memory.cs

```diff
@@ -246,6 +270,46 @@ public abstract partial class OwnedMemory<T> : System.Buffers.IRetainable, Syste
         public abstract void Retain();
         protected internal abstract bool TryGetArray(out System.ArraySegment<T> arraySegment);
     }
+    public readonly partial struct ReadOnlyBuffer<T>
+    {
+        private readonly object _dummy;
+        public static readonly System.Buffers.ReadOnlyBuffer<T> Empty;
+        public ReadOnlyBuffer(System.Buffers.IMemoryList<T> startSegment, int offset, System.Buffers.IMemoryList<T> endSegment, int endIndex) { throw null; }
+        public ReadOnlyBuffer(System.Collections.Generic.IEnumerable<System.Memory<T>> buffers) { throw null; }
+        public ReadOnlyBuffer(System.Memory<T> data) { throw null; }
+        public ReadOnlyBuffer(T[] array) { throw null; }
```

Comment:
I don't remember which of these ctors we agreed to remove when we move stuff to corfx, but I thought we agreed to simplify.

Reply:
We were going to keep the `Memory<T>` and `IMemoryList<T>`

Reply:
All Memory members are private, we need to figure out how to get inner state out of them correctly.